Update `README.md` with initial instructions (#1)
single-cell-data/TileDB-SOMA-Test-Data: README.md

# TileDB-SOMA-Test-Data

This project stores test and example data for TileDB-SOMA. Test data is stored as

## Available data

No data has been added.

## Add new raw datasets

Raw datasets and large binaries should be added to this repository as releases. This is done because we do not need the version control functionality of git for the test data and GitHub has a larger file size limit on the releases then files in the repo.

To add new test data do the following:

1. Create a new release in the repository using the GitHub interface.
2. Pick a title for the dataset (for example "Visium v2.0.0 Dataset") and a description of what the data is. The description may include where the data was originally hosted or how it was otherwise generated.

> 3. Upload the desired datasets to the release by using the file dialog in the release page (at time of writing in Firefox this is done by clicking "Attach binaries by dropping them here or selecting them." in the new release page).

Reviewer: Sorry, I'm still confused. For "Upload the desired datasets" -- what _is_ a desired dataset? What should people be attaching? What should they not be attaching? How will they know if they've attached something correct, or something incorrect?

AFAICT this readme presumes that the reader and the writer both already know (and agree on) what the release artifacts are.

Reviewer: Can you maybe show some examples of what a 'desired dataset' might look like? And maybe what a `tar ztf` of one of those should look like?

Also, how people can test-drive their "desired dataset" before attaching it?

Author: As discussed offline, how to test-drive desired datasets will be added to the TileDB-SOMA side if/when needed.

# TileDB-SOMA-Test-Data

> This project stores test and example data for TileDB-SOMA. Some examples of possible datasets that might be included are:

Reviewer: ```suggestion
This project stores test and example data for the [TileDB-SOMA repository](https://github.com/single-cell-data/TileDB-SOMA). Some examples of possible datasets that might be included are:
```

* an AnnData H5AD file;
* gene expression data and spatial folder from 10x's SpaceRanger (Visium processing software);

> * a set of multiscale images.

Reviewer: A "set" stored as a tar file?

> * a set of multiscale images.

Author: ```suggestion
* a set of multiscale images stored as a tar file.
```

## Available data

> No data has been added.

Reviewer: ```suggestion
* 2025-02-19: No data has been added to date.
```

> ## Add new raw datasets

Reviewer: ```suggestion
## How to add new datasets
```

I don't know what 'raw' is -- you should define the word, or (better), leave it out.

> Raw datasets and large binaries should be added to this repository as releases. A release should include the input data you need to add a desired new test or example to TileDB-SOMA.

Reviewer: Please give an example of what release tags will be like.

* major.minor.micro?
  * If so, following TileDB-SOMA tags?
  * If not, following what?
* If not, then what -- maybe YYYY-MM-DD format?
* Or something else?

To add new test data do the following:

1. Create a new release in the repository using the GitHub interface.
2. Pick a title for the dataset (for example "Visium v2.0.0 Dataset") and a description of what the data is. The description may include where the data was originally hosted or how it was otherwise generated.
3. Upload the desired datasets to the release by using the file dialog in the release page (at time of writing in Firefox this is done by clicking "Attach binaries by dropping them here or selecting them." in the new release page).
4. Publish the release.
5. Add a description of the new data to this README in the "Available data" section. Include the full URL to access the data (for example "https://github.com/single-cell-data/TileDB-SOMA-Test-Data/releases/download/Visium-v2.0.0-Dataset/spatial.tar.gz")

> Note: Data is being added to releases instead of directly to get because we do not need the version control functionality of git for the test data and GitHub has a larger file size limit on releases than files in the repo.

Reviewer: ```suggestion
Note: Data is being added to releases instead of directly to get because we do not need the version-control functionality of Git for the test data, and GitHub has a larger file-size limit for release-artifact files than it does for files committed to source control within the repository.```

> ## Access raw datasets

Reviewer: Please remove/rename/explain "raw"

### General Access

> The data can be downloaded navigating to the release or using wget or curl to access the data from the URL listed in this README.

Reviewer: ```suggestion
The data can be downloaded navigating to the release or using `wget` or `curl` to access the data from the URL listed in this `README.md`.
```

### Add data to TileDB-SOMA

> If a new dataset is needed for testing or examples in the TileDB-SOMA repository you will need to update the `scripts/prepare-test-data.sh`. Instructions on adding new data will be added to `test/README.md`.

Reviewer: I don't see `scripts/prepare-test-data.sh`.

To add new test data do the following:

1. Create a new release in the repository using the GitHub interface.
2. Set the title to be "Dataset {major}.{minor}" (for example "Dataset 3.0"). Most datasets will only need a major version number, but if you need to upload a multi-part dataset use the minor version.

> 3. Set the tag to "dataset-{major}.{minor}" (for example "dataset-3.0").

Reviewer: Only Q remaining is what are the criteria for bumping minor vs bumping major. If I'm uploading one of these -- how do I know which to do?
